crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -205,11 +200,9 @@ pub(crate) enum AllocAtWithoutReplacement {
 
 impl Entity {
     /// Construct an [`Entity`] from a raw `index` value and a non-zero `generation` value.
-    /// Ensure that the generation value is never greater than `0x7FFF_FFFF`.
+    /// Ensure that the masked generation value is never greater than `0x3FFF_FFFF`.
     #[inline(always)]
     pub(crate) const fn from_raw_and_generation(index: u32, generation: NonZeroU32) -> Entity {
-        debug_assert!(generation.get() <= HIGH_MASK);
```

Comment:
The removal of this check seems a bit suspicious

Reply:
Because if we have Entity ID's being used for components with a toggle-able/hide-able bit, then we can't exclude the flags. I could instead do a different check to see if the `PAIR` bit is toggled on instead, so then we can keep that assert.